app/src/main/java/protect/card_locker/ManageGroupCursorAdapter.java

```diff
@@ -0,0 +1,122 @@
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.graphics.BlendModeColorFilterCompat;
+import androidx.core.graphics.BlendModeCompat;
+import androidx.recyclerview.widget.RecyclerView;
+
+import protect.card_locker.preferences.Settings;
+
+public class ManageGroupCursorAdapter extends LoyaltyCardCursorAdapter {
+    private HashMap<Integer, Integer> mIndexCardMap;
+    private HashMap<Integer, Boolean> mInGroupOverlay;
+    private Group mGroup;
+    private DBHelper mDb;
+    public ManageGroupCursorAdapter(Context inputContext, Cursor inputCursor, CardAdapterListener inputListener, Group group){
+        super(inputContext, inputCursor, inputListener);
+        mGroup = new Group(group._id, group.order);
+        mInGroupOverlay = new HashMap<Integer, Boolean>();
+        mDb = new DBHelper(inputContext);
+    }
+
+    @Override
+    public void swapCursor(Cursor inputCursor) {
+        super.swapCursor(inputCursor);
+        mIndexCardMap = new HashMap<Integer, Integer>();
+    }
+
+    @Override
+    public void onBindViewHolder(LoyaltyCardListItemViewHolder inputHolder, Cursor inputCursor){
+        LoyaltyCard loyaltyCard = LoyaltyCard.toLoyaltyCard(inputCursor);
+        Boolean overlayValue = mInGroupOverlay.get(loyaltyCard.id);
+        if((overlayValue != null? overlayValue: isLoyaltyCardInGroup(loyaltyCard.id))) {
+            mAnimationItemsIndex.put(inputCursor.getPosition(), true);
+            mSelectedItems.put(inputCursor.getPosition(), true);
+        }
+        mIndexCardMap.put(inputCursor.getPosition(), loyaltyCard.id);
+        super.onBindViewHolder(inputHolder, inputCursor);
+    }
+
+    private boolean isLoyaltyCardInGroup(int cardId){
+        List<Group> groups = mDb.getLoyaltyCardGroups(cardId);
+        Iterator<Group> groupItr = groups.listIterator();
+        while(groupItr.hasNext()){
+            if (groupItr.next().equals(mGroup)){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void toggleSelection(int inputPosition){
+        super.toggleSelection(inputPosition);
+        int cardId = mIndexCardMap.get(inputPosition);
+        Boolean overlayValue = mInGroupOverlay.get(cardId);
+        if (overlayValue == null){
+            mInGroupOverlay.put(cardId, !isLoyaltyCardInGroup(cardId));
+        }else{
+            mInGroupOverlay.remove(cardId);
+        }
+    }
+
+    public boolean hasChanged() {
+        return mInGroupOverlay.size() > 0;
+    }
+
+    public void commitToDatabase(Context context){
+        // this is very inefficient but done to keep the size of DBHelper low
+        for(Map.Entry<Integer, Boolean> entry: mInGroupOverlay.entrySet()){
+            int cardId = entry.getKey();
+            List<Group> groups = mDb.getLoyaltyCardGroups(cardId);
```

Comment:
Hmm, I wonder if we can make things cleaner by adding a "getGroups" to LoyaltyCardCursor which caches the group list inside the LoyaltyCardCursor, so we don't have to query the DB twice per card (once to see if it's in the group and once to save)

Something like this inside LoyaltyCardCursor:

```
public List<Group> getGroups() {
    if (mGroupsList == null) {
        mGroupsList = mDb.getLoyaltyCardGroups(id);
    }
    return mGroupsList;
}
```